resource_type.go

```diff
@@ -76,12 +82,34 @@ func (t ResourceType) validate(raw []byte) (ResourceAttributes, *errors.ScimErro
 			return ResourceAttributes{}, scimErr
 		}
 
-		attributes[extension.Schema.ID] = extensionAttributes
+		schemaAttributes[extension.Schema.ID] = extensionAttributes
+	}
+
+	attributes := make(map[string]interface{}, len(commonAttributes)+len(schemaAttributes))
+
+	for k, v := range schemaAttributes {
+		attributes[k] = v
+	}
+
+	for k, v := range commonAttributes {
+		attributes[k] = v
 	}
 
 	return attributes, nil
 }
 
+func (t ResourceType) validateCommonAttributes(m map[string]interface{}) (map[string]interface{}, *errors.ScimError) {
```

Comment:
The function name is a little ambiguous. It only validates the `externalId`.

Reply:
You can do the validation by making a `schema.SimpleCoreAttribute` for `externalId`.

Reply:
I've switched this by adding a SimpleCoreAttribute ExternalID to the schema on the fly